crates/bevy_sprite/src/texture_atlas_builder.rs

```diff
@@ -9,36 +9,56 @@ use rectangle_pack::{
 };
 use thiserror::Error;
 
+#[derive(Debug, Error)]
+pub enum TextureAtlasBuilderError {
+    #[error("could not pack textures into an atlas within the given bounds")]
+    NotEnoughSpace,
+}
+
 #[derive(Debug)]
+/// A builder which is used to create a texture atlas from many individual
+/// sprites.
 pub struct TextureAtlasBuilder {
-    pub textures: Vec<Handle<Texture>>,
-    pub rects_to_place: GroupedRectsToPlace<Handle<Texture>>,
-    pub initial_size: Vec2,
-    pub max_size: Vec2,
+    /// The grouped rects which must be placed with a key value pair of a
+    /// texture handle to an index.
+    rects_to_place: GroupedRectsToPlace<Handle<Texture>>,
+    /// The sprite size in pixels.
+    initial_size: Vec2,
+    /// The absolute maximum size of a sprite sheet in pixels.
+    max_size: Vec2,
 }
 
 impl Default for TextureAtlasBuilder {
     fn default() -> Self {
-        Self::new(Vec2::new(256., 256.), Vec2::new(2048., 2048.))
+        Self {
+            rects_to_place: GroupedRectsToPlace::new(),
+            initial_size: Vec2::new(256., 256.),
+            max_size: Vec2::new(2048., 2048.),
+        }
     }
 }
 
-#[derive(Debug, Error)]
-pub enum RectanglePackError {
-    #[error("Could not pack textures into an atlas within the given bounds")]
-    NotEnoughSpace,
-}
+pub type TextureAtlasBuilderResult<T> = Result<T, TextureAtlasBuilderError>;
 
 impl TextureAtlasBuilder {
-    pub fn new(initial_size: Vec2, max_size: Vec2) -> Self {
-        Self {
-            textures: Default::default(),
-            rects_to_place: GroupedRectsToPlace::new(),
-            initial_size,
-            max_size,
-        }
+    /// Constructs a new texture atlas builder.
+    pub fn new() -> Self {
```

Comment:
`#[inline]` this. Not sure about the other ones but this follows @cart's general rule about inlining.